docker/Dockerfile

```diff
@@ -88,13 +113,57 @@ RUN cp /FV3/conf/$configure_file \
     if [ ! -z $compile_option ]; then sed -i "33i $compile_option" \
         /FV3/conf/configure.fv3; fi
 
+COPY FV3/atmos_cubed_sphere /FV3/atmos_cubed_sphere
+
+## Convert FV3GFS sources for serialize image
+##---------------------------------------------------------------------------------
+FROM fv3gfs-environment-serialbox AS fv3gfs-sources-true
+
+ENV FV3_BASE=/FV3/original
+COPY --from=fv3gfs-sources-false /FV3 $FV3_BASE
+COPY --from=fv3gfs-sources-false /stochastic_physics $(FV3_BASE)/../stochastic_physics 
+
+ARG compile_option
+ARG configure_file=configure.fv3.gnu_docker_serialize
+
+# copy appropriate configuration file to configure.fv3
+RUN cp $FV3_BASE/conf/$configure_file \
+        $FV3_BASE/conf/configure.fv3 && \
+    if [ ! -z $compile_option ]; then sed -i "33i $compile_option" \
+        $FV3_BASE/conf/configure.fv3; fi
+
+ENV SERIALBOX_OUTDIR=/FV3
+
+# build model with serialization
+RUN cd $FV3_BASE && make serialize_preprocess
+
+FROM fv3gfs-sources-$serialize AS fv3gfs-sources
+
+## Build FV3 executable in its own image
+##---------------------------------------------------------------------------------
+FROM fv3gfs-environment AS fv3gfs-build
+
+ENV SERIALBOX_DIR=/serialbox2
+ENV SERIALBOX_OUTDIR=/FV3
+ENV PPSER_PY=$(SERIALBOX_DIR)/python/pp_ser/pp_ser.py
```

Comment:
Does Serialbox exist in the default non-serialized image?

Reply:
Now it does, yes. It can be ignored though.